### PR TITLE
POLIO-1702:  pdf preview

### PR DESCRIPTION
**Pull request by beygorghor**
We need to display a preview of pdf on:
- Stock variations (Form A, Destruction Report, Incident Report)
- Supply chain  (VRF, PRE-ALERT)

Related JIRA tickets :  POLIO-1702
## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes

- installed react-pdf
- create a dedicated component to preview a pdf
- create a dedicated component to pdf file upload
- replace it a a bit everywhere

## How to test

Upload pdf files on those pages 
- Stock variations (Form A, Destruction Report, Incident Report)
- Supply chain  (VRF, PRE-ALERT)

Make sure you can make preview all of the pdf

## Print screen / video
Screenshot 2024-11-08 at 14 09 31

Screenshot 2024-11-08 at 14 09 42

### CODE REVIEW

File: plugins/polio/js/src/domains/VaccineModule/SupplyChain/Details/shared.tsx [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
+import { grey } from '@mui/material/colors';
 import { makeStyles } from '@mui/styles';
 import {
     AddButton,
     IntlMessage,
     MENU_HEIGHT_WITH_TABS,
     useSafeIntl,
 } from 'bluesquare-components';
-import { grey } from '@mui/material/colors';
+import React, { FunctionComponent, ReactNode } from 'react';
```
Comment (quang-le) [inline plugins/polio/js/src/domains/VaccineModule/SupplyChain/Details/shared.tsx:10]: I really hate that the linter doesn't leave the React import on top but 🤷‍♂️

File: hat/assets/js/apps/Iaso/components/files/pdf/DocumentUploadWithPreview.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        pdfUrl = document;
+    } else if (
+        Array.isArray(document) &&
+        document.length > 0 &&
+        document[0] instanceof File
+    ) {
+        pdfUrl = URL.createObjectURL(document[0]);
+    } else if (document instanceof File) {
+        pdfUrl = URL.createObjectURL(document[0]);
+    }
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/components/files/pdf/DocumentUploadWithPreview.tsx]: Are you sure this is correct? The 2 `else if` do exactly the same thing

File: hat/assets/js/apps/Iaso/components/files/pdf/PdfPreview.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        defaultMessage: 'Close',
+        id: 'blsq.buttons.label.close',
+    },
+    download: {
+        defaultMessage: 'Download',
+        id: 'blsq.buttons.label.download',
+    },
+});
+
+export const PdfPreview: React.FC<PdfPreviewProps> = ({ pdfUrl }) => {
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/components/files/pdf/PdfPreview.tsx]: Olivier-style typescript, lol

**Review by quang-le — APPROVED** [review]
Just checkout missing translation and the comment about the if else conditions

Screenshot 2024-11-08 at 14 32 17

File: hat/assets/js/apps/Iaso/components/files/pdf/DocumentUploadWithPreview.tsx (hunk 2 above)
Comment (beygorghor) [reply hat/assets/js/apps/Iaso/components/files/pdf/DocumentUploadWithPreview.tsx]: you are right

**Pull request merged**